__tests__/__unit__/abstract/TreeProvider.unit.test.ts

```diff
@@ -0,0 +1,119 @@
+                WorkspaceFolder: 3
+            };
+        })
+    };
+
+    Object.defineProperty(vscode, "ConfigurationTarget", { value: globalMocks.enums, configurable: true });
+    Object.defineProperty(vscode.window, "createTreeView", { value: globalMocks.createTreeView, configurable: true });
+    Object.defineProperty(vscode, "ProgressLocation", { value: globalMocks.ProgressLocation, configurable: true });
+    Object.defineProperty(vscode.window, "withProgress", { value: globalMocks.withProgress, configurable: true });
+    Object.defineProperty(vscode.workspace, "getConfiguration", { value: globalMocks.getConfiguration, configurable: true });
+    Object.defineProperty(Profiles, "getInstance", {
+        value: jest.fn(() => {
+            return {
+                allProfiles: [globalMocks.testProfile, { name: "firstName" }, { name: "secondName" }],
+                getDefaultProfile: globalMocks.mockDefaultProfile,
+                validProfile: ValidProfileEnum.VALID,
+                checkCurrentProfile: jest.fn(),
+                loadNamedProfile: globalMocks.mockLoadNamedProfile,
+                editSession: globalMocks.mockEditSession
+            };
+        }),
+        configurable: true
+    });
+
+    globalMocks.mockAffects.mockReturnValue(true);
+    globalMocks.withProgress.mockImplementation((progLocation, callback) => callback());
+    globalMocks.withProgress.mockReturnValue(globalMocks.testResponse);
+    globalMocks.testSessionNode = createUSSSessionNode(globalMocks.testSession, globalMocks.testProfile);
+    globalMocks.testUSSTree = await createUSSTree(Logger.getAppLogger());
+    Object.defineProperty(globalMocks.testUSSTree, "refresh", { value: globalMocks.refresh, configurable: true });
+    globalMocks.testUSSTree.mSessionNodes.push(globalMocks.testSessionNode);
+    globalMocks.mockLoadNamedProfile.mockReturnValue(globalMocks.testProfile);
+    globalMocks.mockDefaultProfile.mockReturnValue(globalMocks.testProfile);
+    globalMocks.mockEditSession.mockReturnValue(globalMocks.testProfile);
+    globalMocks.getConfiguration.mockReturnValue({
+        get: (setting: string) => [
+            "[test]: /u/aDir{directory}",
+            "[test]: /u/myFile.txt{textFile}",
+        ],
+        update: jest.fn(()=>{
+            return {};
+        })
+    });
+
+    return globalMocks;
+}
+
+describe("ZoweJobNode unit tests - Function editSession", () => {
+    async function createBlockMocks(globalMocks) {
+        const newMocks = {
+            testIJob: createIJobObject(),
+            testJobsProvider: await createJobsTree(Logger.getAppLogger()),
+            jobNode: null
+        };
+
+        newMocks.jobNode = new Job("jobtest", vscode.TreeItemCollapsibleState.Expanded,
+                                   null, globalMocks.testSession, newMocks.testIJob, globalMocks.testProfile);
+        newMocks.jobNode.contextValue = "job";
+        newMocks.jobNode.dirty = true;
+
+        return newMocks;
+    }
```

Comment:
I would prefer the style where we create constants and then return an object composed of them rather than creating an object first and then mutating it.

Reply:
I hope everyone had a chance to read this comment.
I do believe it's personal preference though.
I'm ok either way 😋